### DISCUSSION TITLE
Search for public dockerhub instead of private image registry

### DISCUSSION THREAD

**Discussion by Stef16Robbe** [renovatebot/renovate#29913] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab, Renovate Runner version 17.0.0

## The problem

I host my K8S images in a SaaS Jfrog instance via [remote repositories](https://jfrog.com/help/r/jfrog-artifactory-documentation/remote-docker-repositories). These are configured through different URLs. I was wondering if/how it's possible to configure Renovate to search on the remote repository when finding certain URLs instead of on the private registry, like so:

- `foo.jfrog.io/public-dockerhub/{image}` should look at `index.docker.io/v2/{image}`
- `foo.jfrog.io/public-ghcr/{image}` should look at `ghcr.io/{image}`
- etc...

This would fix two things:

1. I don't need authentication which I've found troublesome to set-up securely.
2. More importantly, the private registry only contains the images that have been pulled, and we obviously only pull hard-coded versions, so even with proper authentication Renovate won't see any updated images because they are only pulled when we update the versions on our end.

So it shouldn't replace the image in code, just use the alternative URL to find any updated versions.

## What I tried

Configuring package rules:

```json
...
    "packageRules": [
        {
            "matchPackagePrefixes": [
                "foo.jfrog.io/public-dockerhub"
            ],
            "registryUrls": [
                "https://index.docker.io/v2",
                "https://ghcr.io"
            ]
    ]
...
```

Then I see these warnings in the logs:

```plaintext
 WARN: Excess registryUrls found for datasource lookup - using first configured only (repository=foo)
       "datasource": "docker",
       "packageName": "docker.io/{image}",
       "registryUrls": ["https://docker.io", "https://hub.docker.com", "https://ghcr.io"]
```

And then the debug logs show it's still trying jfrog.io:

```
DEBUG: Unauthorized docker lookup (repository=foo)
       "registryHost": "https://foo.jfrog.io",
       "dockerRepository": "public-dockerhub/{image}"
DEBUG: Response code 401 (Unauthorized)
```

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by Stef16Robbe** [comment]
I've also tried to use a `customManager`:

```json
...
    "customManagers": [
        {
            "customType": "regex",
            "fileMatch": [
                "\\.yaml$"
            ],
            "matchStrings": [
                "(?<depName>foo\\.jfrog\\.io/public-dockerhub/aRepository/andImage):(?<currentValue>[^\\s]+)"
            ],
            "depNameTemplate": "docker.io/aRepository/andImage",
            "datasourceTemplate": "docker"
        }
    ]
...
```

But I'm not sure if I'm using this correctly, logs are still the same showing attempts towards `jfrog.io`.

**Comment by rarkins** [comment, marked as answer]
Try `registryAliases`

**Reply by Stef16Robbe** [reply]
Awesome, thanks very much! Just completely missed that. For anyone curious, this is what it'll look like:

```json
"registryAliases": {
    "foo.jfrog.io/public-dockerhub": "index.docker.io/v2",
    "foo.jfrog.io/public-ghcr": "ghcr.io"
} 
```
